src/main/java/com/commercetools/sync/categories/CategorySync.java

```diff
@@ -0,0 +1,190 @@
+                    getStatistics().incrementProcessed();
+                    final String externalId = categoryDraft.getExternalId();
+                    if (isNotBlank(externalId)) {
+                        createOrUpdateCategory(categoryDraft);
+                    } else {
+                        failSync(format("CategoryDraft with name: %s doesn't have an externalId.",
+                                categoryDraft.getName().toString()), null);
+                    }
+                }
+            } catch (ClassCastException e) {
+                failSync(format("Element at position %d is not an instance of CategoryDraft.", i), e);
+            }
+
+        }
+        getStatistics().calculateProcessingTime();
+    }
+
+    /**
+     * Given a category draft {@link CategoryDraft} with an externalId, this method tries to fetch the existing category
+     * from CTP project stored in the {@code syncOptions} instance of this class. It then either creates a new category,
+     * if none exist with the same external id, or update the existing category.
+     * <p>
+     * The {@code statistics} instance is updated accordingly to whether the CTP request was carried out successfully or not.
+     * If an exception was thrown on executing the request to CTP,
+     * the optional error callback specified in the {@code syncOptions} is called.
+     *
+     * @param categoryDraft the category draft where we get the new data.
+     */
+    void createOrUpdateCategory(@Nonnull final CategoryDraft categoryDraft) {
+        final String externalId = categoryDraft.getExternalId();
+        try {
+            final Category oldCategory = getSyncOptions().getCategoryService().fetchCategoryByExternalId(externalId);
+            if (oldCategory != null) {
+                syncCategories(oldCategory, categoryDraft);
+            } else {
+                createCategory(categoryDraft);
+            }
+        } catch (SphereException e) {
+            failSync(format("Failed to fetch category with external id" +
+                            " '%s' in CTP project with key '%s",
+                    externalId, getSyncOptions().getClientConfig().getProjectKey()), e);
+        }
+    }
+
+    /**
+     * Given a {@link CategoryDraft}, this method issues a blocking request to the CTP project defined by the client
+     * configuration stored in the {@code syncOptions} instance of this class to create a category with the same fields as
+     * this category draft.
+     * <p>
+     * The {@code statistics} instance is updated accordingly to whether the CTP request was carried out successfully or not.
+     * If an exception was thrown on executing the request to CTP,
+     * the optional error callback specified in the {@code syncOptions} is called.
+     *
+     * @param categoryDraft the category draft to create the category from.
+     */
+    private void createCategory(@Nonnull final CategoryDraft categoryDraft) {
```

Comment:
These fields are checked when comparing the fields of the new draft to it's adjacent category. You could have two empty fields for example description then it's not required. However, if it was already set in the existing one and not set in the new one, then it's required in this case. So we have to check for required fields later on field-2-field comparison.